sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/parameters.scala

```diff
@@ -189,7 +189,8 @@ object BindParameters extends ParameterizedQueryProcessor with QueryErrorsBase {
       // We should wait for `CTESubstitution` to resolve CTE before binding parameters, as CTE
       // relations are not children of `UnresolvedWith`.
```

Comment:
I'm thinking of a different approach: shall we explicitly match `UnresolvedWith` here and bind parameters in the CTE relations? Then we no longer need this hack to delay parameter binding.

Reply:
Can we do that in 2 steps: 1. fix the issue, 2. do the proposed refactoring? It seems less risky, so, the first patch could be backported more safely.

Reply:
+1